akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -80,97 +82,84 @@ import akka.stream.scaladsl.Source
-      handlerStrategy)
+    copy(offsetStrategy = atLeastOnceStrategy
+      .copy(afterEnvelopes = Some(afterEnvelopes), orAfterDuration = Some(afterDuration.toScala)))
 
   /**
    * Settings for GroupedCassandraProjection
    */
   override def withGroup(
       groupAfterEnvelopes: Int,
       groupAfterDuration: FiniteDuration): CassandraProjectionImpl[Offset, Envelope] =
-    new CassandraProjectionImpl(
-      projectionId,
-      sourceProvider,
-      settingsOpt,
-      offsetStrategy,
-      handlerStrategy
-        .asInstanceOf[GroupedHandlerStrategy[Envelope]]
-        .copy(afterEnvelopes = Some(groupAfterEnvelopes), orAfterDuration = Some(groupAfterDuration)))
+    copy(handlerStrategy = handlerStrategy
+      .asInstanceOf[GroupedHandlerStrategy[Envelope]]
+      .copy(afterEnvelopes = Some(groupAfterEnvelopes), orAfterDuration = Some(groupAfterDuration)))
 
   /**
    * Java API
    */
   override def withGroup(
       groupAfterEnvelopes: Int,
       groupAfterDuration: java.time.Duration): CassandraProjectionImpl[Offset, Envelope] =
-    new CassandraProjectionImpl(
-      projectionId,
-      sourceProvider,
-      settingsOpt,
-      offsetStrategy,
-      handlerStrategy
-        .asInstanceOf[GroupedHandlerStrategy[Envelope]]
-        .copy(afterEnvelopes = Some(groupAfterEnvelopes), orAfterDuration = Some(groupAfterDuration.toScala)))
+    copy(handlerStrategy = handlerStrategy
+      .asInstanceOf[GroupedHandlerStrategy[Envelope]]
+      .copy(afterEnvelopes = Some(groupAfterEnvelopes), orAfterDuration = Some(groupAfterDuration.toScala)))
 
   override def withRecoveryStrategy(
       recoveryStrategy: HandlerRecoveryStrategy): CassandraProjectionImpl[Offset, Envelope] =
-    new CassandraProjectionImpl(
-      projectionId,
-      sourceProvider,
-      settingsOpt,
-      atLeastOnceStrategy.copy(recoveryStrategy = Some(recoveryStrategy)),
-      handlerStrategy)
+    copy(offsetStrategy = atLeastOnceStrategy.copy(recoveryStrategy = Some(recoveryStrategy)))
 
   /**
    * Settings for AtMostOnceCassandraProjection
    */
   override def withRecoveryStrategy(
       recoveryStrategy: StrictRecoveryStrategy): CassandraProjectionImpl[Offset, Envelope] =
-    new CassandraProjectionImpl(
-      projectionId,
-      sourceProvider,
-      settingsOpt,
-      atMostOnceStrategy.copy(recoveryStrategy = Some(recoveryStrategy)),
-      handlerStrategy)
+    copy(offsetStrategy = atMostOnceStrategy.copy(recoveryStrategy = Some(recoveryStrategy)))
+
+  override def withStatusObserver(observer: StatusObserver[Envelope]): CassandraProjectionImpl[Offset, Envelope] =
+    copy(statusObserver = observer)
 
   /**
    * INTERNAL API
    * Return a RunningProjection
    */
   @InternalApi
-  override private[projection] def run()(implicit system: ActorSystem[_]): RunningProjection =
+  override private[projection] def run()(implicit system: ActorSystem[_]): RunningProjection = {
+    statusObserver.started(projectionId)
```

Comment:
might be better to call the started inside `mappedSource` (see comment bellow on testing with SinkProbe

Reply:
fixed